### DISCUSSION TITLE
Implement div_euclid / rem_euclid for vec types

### DISCUSSION THREAD

**Discussion by SpecificProtagonist** [bitshifter/glam-rs#417] [Ideas]
Rust's float and signed integer types implement euclidian division via the `div_euclid` / `rem_euclid` methods, but this is missing from glam's vec types (in my case, I need it to divide an area into equally sized chunks).

Edit: made a non-vectorized implementation in #419

**Comment by SpecificProtagonist** [comment]
Looks like this came up in #218, but only `div` / `rem` were implemented. I'm a bit confused about the comment, as Rust's `div_euclid` / `rem_euclid` were stabilized in [before that issue was opened](https://github.com/rust-lang/rust/pull/61884#event-2511286186).

**Reply by bitshifter** [reply]
That was the stabilization commit, those methods didn't make it into a stable Rust release until 1.38.0 in September 2019 https://releases.rs/docs/1.38.0/

**Reply by SpecificProtagonist** [reply]
Yeah, but that's still a much older Rust version than the MSRV at the time (*shrug*, doesn't matter now).

**Reply by bitshifter** [reply]
Hmm, yeah I'm not sure why I didn't think they weren't stabilized in 2021, other than the methods appear to have been renamed since the RFC that Alice linked so maybe I didn't find them in the docs or something.

**Comment by bitshifter** [comment]
These have been added now.
